### ISSUE TITLE
ppl:nan after step5000

### ISSUE THREAD

**Issue by txAnnie** [OpenNMT/OpenNMT-py#1072]
[2018-11-26 14:47:26,759 INFO] Step 5350/50000; acc:  51.25; ppl:  8.43; xent: 2.13; lr: 0.00066; 1320/1445 tok/s;  19995 sec
[2018-11-26 14:50:30,698 INFO] Step 5400/50000; acc:  48.90; ppl:  8.91; xent: 2.19; lr: 0.00067; 1628/1740 tok/s;  20179 sec
[2018-11-26 14:53:33,585 INFO] Step 5450/50000; acc:  48.07; ppl:  9.73; xent: 2.28; lr: 0.00067; 2081/2351 tok/s;  20362 sec
[2018-11-26 14:56:55,137 INFO] Step 5500/50000; acc:  45.16; ppl: 11.12; xent: 2.41; lr: 0.00068; 2132/1928 tok/s;  20564 sec
[2018-11-26 14:59:57,699 INFO] Step 5550/50000; acc:  47.00; ppl: 10.46; xent: 2.35; lr: 0.00069; 1851/1358 tok/s;  20746 sec
[2018-11-26 15:02:57,162 INFO] Step 5600/50000; acc:  47.14; ppl:  9.83; xent: 2.29; lr: 0.00069; 1065/1591 tok/s;  20926 sec
[2018-11-26 15:06:01,481 INFO] Step 5650/50000; acc:   0.19; ppl:   nan; xent:  nan; lr: 0.00070; 1928/2310 tok/s;  21110 sec
[2018-11-26 15:09:00,598 INFO] Step 5700/50000; acc:   0.17; ppl:   nan; xent:  nan; lr: 0.00070; 1827/2180 tok/s;  21289 sec
[2018-11-26 15:09:03,577 INFO] Loading train dataset from ./disc_data/data.train.1.pt, number of examples: 221287
[2018-11-26 15:12:08,893 INFO] Step 5750/50000; acc:   0.12; ppl:   nan; xent:  nan; lr: 0.00071; 1664/2365 tok/s;  21477 sec

the command is :CUDA_VISIBLE_DEVICES=3 nohup python3 train.py -data ./disc_data/data -save_model ./disc_data/model -layers 6 -rnn_size 512 -word_vec_size 512 -transformer_ff 2048 -heads 8 -encoder_type transformer -decoder_type transformer -position_encoding -train_steps 50000 -max_g    enerator_batches 2 -dropout 0.1 -batch_size 4096 -batch_type tokens  -normalization tokens -accum_count 2 -optim adam -adam_beta2 0.998  -decay_method noam -warmup_steps 8000 -learning_rate 2 -max_grad_norm 0 -param_init 0 -param_init_glorot -label_smoothing 0.1 -valid_ steps 2000 -save_checkpoint_steps 2000 -gpuid 3 > log.hierarchical_ test &

what caused this issue and how can i fix it ?

**Comment by vince62s** [comment]
you are not on master.
But my first thought is that there is some issue with your data.

**Comment by txAnnie** [comment]
But i didn't meet this issue before when using the same data, So i guess it's not be my data problem, what else can cause this issue?

**Comment by vince62s** [comment]
please pull master because without knowing your hash version we cannot be sure where you are.

**Comment by zengyan-97** [comment]
> I have fixed the issue, thanks so much

I have the same issue. How did you fix it?
